### PR TITLE
feat(hatchery): region prerequisite

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: docs/content/docs/concepts/requirement/requirement_region.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: "Region Requirement"
+weight: 4
+---
+
+The `Region` prerequisite allows you to require a worker to have access to a specific region.
+
+A `Region` can be configured on each hatchery. With a free text as `myregion` in hachery configuration,
```
Comment (richardlt) [inline docs/content/docs/concepts/requirement/requirement_region.md:8]:
```suggestion
A `Region` can be configured on each hatchery. With a free text as `myregion` in hatchery configuration, 
```

File: engine/api/workermodel/registration.go [hunk 2]
```diff
@@ -59,6 +59,9 @@ func ComputeRegistrationNeeds(db gorp.SqlExecutor, allBinaryReqs sdk.Requirement
 	if nbHostnameReq > 1 {
 		return sdk.NewError(sdk.ErrWrongRequest, errors.New("invalid hostname requirement usage"))
 	}
+	if nbRegionReq > 1 {
+		return sdk.NewError(sdk.ErrWrongRequest, errors.New("invalid region requirement usage"))
```
Comment (richardlt) [inline engine/api/workermodel/registration.go:63]:
```suggestion
		return sdk.NewErrorFrom(sdk.ErrWrongRequest, "invalid region requirement usage")
```

File: engine/api/workermodel/registration.go [hunk 3]
```diff
@@ -59,6 +59,9 @@ func ComputeRegistrationNeeds(db gorp.SqlExecutor, allBinaryReqs sdk.Requirement
 	if nbHostnameReq > 1 {
 		return sdk.NewError(sdk.ErrWrongRequest, errors.New("invalid hostname requirement usage"))
```
Comment (richardlt) [inline engine/api/workermodel/registration.go:60]:
```suggestion
		return sdk.NewErrorFrom(sdk.ErrWrongRequest, "invalid hostname requirement usage")
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13108.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✘  
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13115.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13115.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Pull request merged**